packages/actor-rdf-resolve-quad-pattern-hypermedia/test/MediatedLinkedRdfSourcesAsyncRdfIterator-test.ts

```diff
@@ -101,6 +101,12 @@ describe('MediatedLinkedRdfSourcesAsyncRdfIterator', () => {
       );
     });
 
+    afterEach(() => {
+      // This is here since the MediatedLinkedRdfSourcesAsyncRdfIterator is never fully consumed
+      // **THIS IS AN ALARM BELL** do not merge without checking this
+      source.destroy();
+    });
+
```

Comment:
I think there is likely to be a more fundamental problem with the MediatedLinkedRdfSourcesAsyncRdfIterator here because it is always leaving iterators open.

Reply:
Yeah, this is a complex part of this codebase, so there might be issues with it.
Something you want to look into within the scope of this PR?